hat/assets/js/apps/Iaso/domains/orgUnits/reviewChanges/Filter/ReviewOrgUnitChangesFilter.tsx

```diff
@@ -82,6 +145,52 @@ export const ReviewOrgUnitChangesFilter: FunctionComponent<Props> = ({
         [handleChange],
     );
 
+    const handleDataSourceVersionChange = useCallback(
+        (key, newValue) => {
+            if (key === 'source') {
+                setDataSource(newValue);
+                const selectedSource = dataSources?.filter(
+                    source => source.value === newValue,
+                )[0];
+                setSelectedVersionId(
+                    selectedSource?.original?.default_version.id.toString(),
+                );
+                handleChange(
+                    'source_version_id',
+                    selectedSource?.original?.default_version.id,
+                );
+            } else {
+                setSelectedVersionId(newValue.toString());
+                handleChange('source_version_id', newValue);
```

Comment:
I think you need additional code here to reset the treeview when changing source/version